distsql/distsql.go

```diff
@@ -136,16 +138,20 @@ func (r *selectResult) Next(goCtx goctx.Context) (PartialResult, error) {
 	pr.rowLen = r.rowLen
 	err := pr.unmarshal(re.result)
 	if len(pr.resp.OutputCounts) > 0 {
-		r.scanCount += pr.resp.OutputCounts[0]
+		scanKeysPartial := pr.resp.OutputCounts[0]
+		metrics.DistSQLScanKeysPartialHistogram.Observe(float64(scanKeysPartial))
+		r.scanKeys += scanKeysPartial
 	} else {
-		r.scanCount = -1
+		r.scanKeys = -1
 	}
+	r.partialCount++
 	return pr, errors.Trace(err)
 }
 
 // NextRaw returns the next raw partial result.
 func (r *selectResult) NextRaw(goCtx goctx.Context) ([]byte, error) {
 	re := <-r.results
+	r.partialCount++
```

Comment:
Should we set the `scanKeys` to `-1`, because it will never update the `scanKeys`?